### PR TITLE
[#11357] Allow access to submission/results page for unregistered instructors

### LINKED ISSUE #11357: Allow access to submission/results page for unregistered instructors
- **Environment**: live server at V8.0.2, but feature is absent since at least V6.0.0

**Description of feature/enhancement**

As part of https://github.com/TEAMMATES/teammates/projects/12; to be done after #11356 and #11425

Allow instructors to access submission/results page without registration (i.e. only with key).

Note that this does not apply to course-wide results page; that page is correctly locked behind registration and shall remain so.

**Justification**

Submission and result pages are accessible to students without registration, with option for students to register should they choose to. Given that the required functions are almost identical, there is no reason why instructors should not be granted the same flexibility.

**Existing similar features and their shortcomings**

None.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #11357

### CODE REVIEW

File: src/main/java/teammates/logic/api/EmailGenerator.java [hunk 1]
```diff
@@ -760,7 +739,6 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructors(
                 "${submitUrl}", "{in the actual email sent to the students, this will be the unique link}",
                 "${reportUrl}", "{in the actual email sent to the students, this will be the unique link}",
```
Comment (samuelfangjw) [inline src/main/java/teammates/logic/api/EmailGenerator.java:740]: Should these be the actual links instead?

**Review by samuelfangjw — COMMENTED** [review]
Sorry didn't manage to finish reviewing this PR today, will continue on friday. Have two comments i'm not too sure about below.

File: src/main/java/teammates/ui/webapi/GetSessionResultsAction.java [hunk 2]
```diff
@@ -79,7 +94,7 @@ public JsonResult execute() {
             return new JsonResult(SessionResultsData.initForInstructor(bundle));
         case INSTRUCTOR_RESULT:
             // Section name filter is not applicable here
-            instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+            instructor = getInstructor(courseId);
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/main/java/teammates/ui/webapi/GetSessionResultsAction.java:97]: On a normal flow, `instructor == null` will not even pass access control check. What you mentioned is theoretically possible as there is a way to bypass access control check, but not the highest of priorities.

You can check the switch case right before this block (`case FULL_DETAIL`).

**Review by samuelfangjw — COMMENTED** [review]
Some comments below to clarify, the rest looks good to me!

File: src/test/java/teammates/ui/webapi/GetSessionResultsActionTest.java [hunk 3]
```diff
@@ -87,45 +87,53 @@ protected void testExecute() {
...
+        verifyAccessibleForInstructorsOfTheSameCourse(submissionParams);
+        verifyInaccessibleForInstructorsOfOtherCourses(submissionParams);
+
+        ______TS("accessible for authenticated student when published");
+        submissionParams = new String[] {
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, publishedFeedbackSession.getFeedbackSessionName(),
+                Const.ParamsNames.COURSE_ID, publishedFeedbackSession.getCourseId(),
                 Const.ParamsNames.INTENT, Intent.STUDENT_RESULT.name(),
         };
         verifyAccessibleForStudentsOfTheSameCourse(submissionParams);
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/test/java/teammates/ui/webapi/GetSessionResultsActionTest.java:125]: No harm in doing that, I would say

File: src/main/java/teammates/logic/api/EmailGenerator.java [hunk 4]
```diff
@@ -760,7 +739,6 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructors(
...
                 "${feedbackAction}", feedbackAction,
-                "${additionalNotes}", "",
                 "${additionalContactInformation}", additionalContactInformation);
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/main/java/teammates/logic/api/EmailGenerator.java:743]: I might need to check further on this

[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/main/java/teammates/logic/api/EmailGenerator.java:735]: @samuelfangjw I have concluded that it is a problem already existing in the system, not introduced by this PR. This should be a separate extension to the system.

**Review by samuelfangjw — APPROVED** [review]
Looks good to me!

**Pull request merged**